src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -791,9 +791,10 @@ public void BuildSymbolDefinitionMap(ObjectNode node, ISymbolDefinitionNode[] de
 
         private void AppendExternCPrefix(Utf8StringBuilder sb)
         {
-            if (_targetPlatform.OperatingSystem == TargetOS.OSX)
+            if (_targetPlatform.IsOSX ||
```

Comment:
We might want to get rid of `AppendExternCPrefix` and move everything into the NodeMangler.